core/HoistComponent.js

```diff
@@ -264,7 +264,7 @@ function lookupModel(spec, props, modelLookup, displayName) {
 
     // 2) props - instance
     if (model) {
-        if (!matchesSelector(model, selector, true)) {
+        if (!model.isHoistModel || !model.matchesSelector(selector, true)) {
```

Comment:
Should we split these out into separate checks so we can provide a more relevant error message below?